test-upgrade/reputation-mining-cycle-upgrade.js

```diff
@@ -0,0 +1,46 @@
+/* globals artifacts */
+/* eslint-disable prefer-arrow-callback */
+import { setupReputationMiningCycleResolver } from "../helpers/upgradable-contracts";
+import { advanceMiningCycleNoContest } from "../helpers/test-helper";
+
+const IColonyNetwork = artifacts.require("IColonyNetwork");
+const EtherRouter = artifacts.require("EtherRouter");
+const Resolver = artifacts.require("Resolver");
+const UpdatedReputationMiningCycle = artifacts.require("./UpdatedReputationMiningCycle");
+const ReputationMiningCycleRespond = artifacts.require("./ReputationMiningCycleRespond");
+
+contract("ReputationMiningCycle contract upgrade", function() {
+  let colonyNetwork;
+  let reputationMiningResolverBefore;
+
+  before(async function() {
+    const etherRouter = await EtherRouter.deployed();
+    colonyNetwork = await IColonyNetwork.at(etherRouter.address);
+    reputationMiningResolverBefore = await colonyNetwork.getMiningResolver();
+
+    // Register a new version of the ReputationMiningCycleResolver on the ColonyNetwork
+    const reputationMiningCycle = await UpdatedReputationMiningCycle.new();
+    const reputationMiningCycleRespond = await ReputationMiningCycleRespond.deployed();
+
+    const resolver = await Resolver.new();
+    await resolver.register("isUpdated()", reputationMiningCycle.address);
+    await setupReputationMiningCycleResolver(reputationMiningCycle, reputationMiningCycleRespond, resolver, colonyNetwork);
+  });
+
+  describe("when upgrading ReputationMiningCycle contract", function() {
+    it("should change the ReputationMiningCycle resolver on the Network", async function() {
+      const reputationMiningResolverAfter = await colonyNetwork.getMiningResolver();
+      assert.notEqual(reputationMiningResolverBefore, reputationMiningResolverAfter);
+    });
+
+    it("should pick up the new ReputationMiningCycle contract", async function() {
+      // Cycle through the mining process twice to get the newly created inactive cycles on the updated mining contract
```

Comment:
Once?

Reply:
Indeed. I started off with going through it twice but there's no point.